### ISSUE TITLE
UI: Links to Sub-applications in other namespace are broken

### ISSUE THREAD

**Issue by iandelahorne** [argoproj/argo-cd#11945]
**Describe the bug**
At Patreon we use the App-of-Apps pattern combined with "Applications in other namespaces" (#9755 ) for teams to manage applications in their own namespaces. 
However, the UI needs a slight update to make it easy to follow links from the AoA to the sub-application. Currently the path in the  "Open application" link in the tree or list view does not contain the sub-application's namespace.

**To Reproduce**

1. Create an App-of-Apps which creates applications in namespaces other than `argocd`. 
Example in [iandelahorne/argocd-example-apps/apps, namespace branch](https://github.com/iandelahorne/argocd-example-apps/tree/namespace/apps) which creates the applications in the `ns` namespace.
2. Navigate to the application in the UI and sync.
3. Attempt to follow the `Open Application` link in the tree or list view 
![Screenshot 2023-01-10 at 18 11 59](https://user-images.githubusercontent.com/184966/211702013-a43af1ab-1fac-4be1-85ba-f3e6e93122c3.png)

4. This will link to `/applications/helm-guestbook`, which 404's
5. Navigating to `/applications/ns/helm-guestbook` manually, however, will work.

**Expected behavior**
The `Open application` link to a sub-application in an app-of-app should include the app's namespace in the path.

**Screenshots**

Tile view of UI, with [iandelahorne/argocd-example-apps/apps, namespace branch](https://github.com/iandelahorne/argocd-example-apps/tree/namespace/apps) application added as Application `aoa`
![Screenshot 2023-01-10 at 18 16 38](https://user-images.githubusercontent.com/184966/211703616-a23e9509-58c8-4991-a673-6d7b97717ed1.png) 

Hovering over link to `helm-guestbooks` application, showing lack of namespace in path
![Screenshot 2023-01-10 at 18 22 32](https://user-images.githubusercontent.com/184966/211703684-4bd39ccf-74eb-4ef0-8c4c-d1006714a277.png)

Following link showing a 404.
![Screenshot 2023-01-10 at 18 22 43](https://user-images.githubusercontent.com/184966/211703845-43c1194b-9af3-4751-830f-902179437b66.png)

**Version**
This is also present on latest `master`, but I've added the version in our staging cluster where this exhibits itself.

```shell
$ argocd version
argocd: v2.5.5+fc3eaec.dirty
  BuildDate: 2022-12-16T18:35:58Z
  GitCommit: fc3eaec6f498ddbe49a5fa9d215a219191fba02f
  GitTreeState: dirty
  GoVersion: go1.19.4
  Compiler: gc
  Platform: darwin/arm64
argocd-server: v2.5.2+148d8da
  BuildDate: 2022-11-07T16:42:47Z
  GitCommit: 148d8da7a996f6c9f4d102fdd8e688c2ff3fd8c7
  GitTreeState: clean
  GoVersion: go1.18.8
  Compiler: gc
  Platform: linux/amd64
  Kustomize Version: v4.5.7 2022-08-02T16:35:54Z
  Helm Version: v3.10.1+g9f88ccb
  Kubectl Version: v0.24.2
  Jsonnet Version: v0.18.0
  ```

**Comment by mikeeq** [comment]
I raised that issue over a month ago: https://github.com/argoproj/argo-cd/issues/11518

**Comment by alexef** [comment]
Closed this one as a duplicate of #11518 . Hopefully #11946 fixes it in time for 2.6

**Comment by iandelahorne** [comment]
Apologies @mikeeq, I have no idea why I didn't find #11518 when trying to see if someone else had run into this as well. Thanks for flagging!
